### PR TITLE
OPHJOD-830: Populate favorites page with job opportunities

### PR DESCRIPTION
**Pull request by juhaniko**
## Description

* Populated profile/favorites page with paginated job opportunities
* Removed "Poista valitut ammatit" button

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-830

image

### CODE REVIEW

File: src/i18n/fi/translation.json [hunk 1]
```diff
@@ -205,7 +205,10 @@
     },
     "favorites": {
       "description": "Suosikeiksi tallentamasi koulutus- ja työmahdollisuudet löytyvät täältä. Voit tallentaa lisää mahdollisuuksia suosikeiksi kohtautuskoneen tulosnäkymässä. Voit poistaa suosikkeja sekä luoda yhdestä tai useammasta suosikista koulutus- ja urapolkusuunnitelman.",
-      "title": "Suosikkini"
+      "title": "Suosikkini",
+      "you-have-saved-n-opportunities_one": "Olet tallentanut yhden mahdollisuuden suosikiksi",
```
Comment (juhaniko) [inline src/i18n/fi/translation.json:209]: Tried the plurals feature from i18next, I think it makes things a bit more user friendly. Any opinions about this?

Comment (ketsappi) [reply src/i18n/fi/translation.json:209]: I find this great feature to be used 👍

File: src/routes/Profile/Favorites/loader.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { components } from '@/api/schema';
+import { useSuosikitStore } from '@/stores/useSuosikitStore';
+import { LoaderFunction } from 'react-router-dom';
+
+export default (async () => {
+  const { fetchSuosikit, fetchPage } = useSuosikitStore.getState();
+  await fetchSuosikit();
+  await fetchPage({ page: 1 });
+  return null;
```
[the remark this answers is not in the document]
Comment (juhaniko) [reply src/routes/Profile/Favorites/loader.ts:9]: This was the issue I mentioned in chat earlier. I decided to try the approach where everything data ralated is handled in the zustand store. Now the loader is just for triggering data fetching, but the actual data processing happens in the store functions.

[the remark this answers is not in the document]
Comment (juhaniko) [reply src/routes/Profile/Favorites/loader.ts:9]: The store could be ommited and have (almost) all of the logic in the component, but that would result in a bit of code duplication: Pretty much the same `client.GET` calls in the loader for initial data,  and also in the component after page changes/favorite deletions.

My reasoning for using the store was to eliminate any code duplication and have the business logic be in one place.

**Review by sauanto — APPROVED** [review]

**Pull request merged**